### DISCUSSION TITLE
Assertion Failed!

### DISCUSSION THREAD

**Discussion by Djard** [hydrogen-music/hydrogen#1536] [Hydrogen on Windows]
I can play back patterns; but in song mode, the notice below appears, after which the program automatically shuts down. Restarting and reinstalling Hydrogen did not help. I also tried re-installing version 2015 of Visual C++ Redistributable; but the failure notice keeps appearing, whenever I try to play back my song. The song played properly before Microsoft updated my Win 10 system, though looking at the history of updates, Visual C++ was not among them.

![Assertion Failed](https://user-images.githubusercontent.com/99554465/159094001-b1c5746b-7ccd-4b37-9476-43d8e0aa3c41.jpg)

Anything else I can try? If not, is there a way to save my work?

**Comment by Djard** [comment]
Never mind. I figured out what caused the problem. Inadvertently, I renamed an audio file that was associated with Hydrogen. Changing the name back to the original did the trick.

**Reply by theGreatWhiteShark** [reply]
Good to head. But it still would be nice if Hydrogen would just panic and open an error popup instead of being terminated by an assertion.
